File: programmers/lv2/기출문제/택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    answer = 0
    
    deliveries, pickups = deliveries[::-1], pickups[::-1]
    d_cap, p_cap = 0, 0
    
    for i in range(n):
        d_cap += deliveries[i]
        p_cap += pickups[i]
        
        while d_cap > 0 or p_cap > 0:
            d_cap -= cap
            p_cap -= cap
            answer += (n - i) * 2
            
    return answer

File: programmers/lv2/기출문제/택배 배달과 수거하기.java
class Solution {
    public long solution(int cap, int n, int[] deliveries, int[] pickups) {
        long answer = 0;
        
        int dCap = 0, pCap = 0;
        for(int i = n - 1; i >= 0; i--) {
            dCap += deliveries[i];
            pCap += pickups[i];
            
            while(dCap > 0 || pCap > 0) {
                dCap -= cap;
                pCap -= cap;
                answer += (i + 1) * 2;
            }
        }
        
        return answer;
    }
}
